partner_resource_delivery_schedule/models/res_partner.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _name = "res.partner"
+    _inherit = ["res.partner", "resource.mixin"]
+
+    resource_id = fields.Many2one(
+        string="Availability Resource",
+        domain="[('resource_type', '=', 'contact')]",
+        required=False,
+    )
```

Comment:
```suggestion
    )
```
Maybe you can add a M2O to resource.calendar, when the calendar resource changes, update the resource. This way it is easier for the user to modify the available resource like Odoo does on the employee.